### PR TITLE
CMake cleanups

### LINKED ISSUE #390: Overhaul/modernize CMake files
* Instead of setting warnings globally and then disabling them for 3rd party code, define a target which configures warnings, and make only our own code depend on that target
* For 3rd party libs where we have our own CMake file, define targets in a way that it's sufficient to depend on those targets (i.e., it should not be necessary to separately specify an include dir or similar)

### PR DESCRIPTION
**Pull request by lethal-guitar**
Closes #390

### CODE REVIEW

**Comment by lethal-guitar** [comment]
@McMartin just in case you are bored and feel like reviewing some CMake, let me know if you notice anything that could be improved further 🙂

**Comment by McMartin** [comment]
@lethal-guitar I have a look now 👀

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -359,6 +359,7 @@ if("${CMAKE_SYSTEM_NAME}" STREQUAL "Emscripten")
       rigel_core
     )
     target_link_options(${main_exe_name} PRIVATE
```
Comment (McMartin) [inline src/CMakeLists.txt]: `target_link_options` was introduced in CMake 3.13. You should update the version in the call to `cmake_minimum_required` in the top-level `CMakeLists.txt` file.

File: CMakeLists.txt [hunk 2]
```diff
@@ -34,7 +34,10 @@ if("${CMAKE_SYSTEM_NAME}" STREQUAL "Emscripten")
         message(STATUS "Bundling game data from path: ${WEBASSEMBLY_GAME_PATH}")
     endif()
 
-    add_compile_options(-O3)
+    if("${CMAKE_BUILD_TYPE}" STREQUAL "Release")
+        # Emscripten's CMake toolchain defaults to O2, but we want O3
+        add_compile_options(-O3)
+    endif()
```
Comment (McMartin) [inline CMakeLists.txt:39]:
```suggestion
    # Emscripten's CMake toolchain defaults to O2, but we want O3
    add_compile_options($<$<NOT:$<CONFIG:Debug>>:-O3>)
```
So it also works with multi-configuration generators, and the `MinSizeRel` and `RelWithDebSymbols` configurations.

File: cmake/rigel.cmake [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+# set the appropriate compiler and linker flags for Emscripten instead of
+# referring to external libraries.
+function(rigel_define_wasm_targets_for_dependencies)
+    add_library(SDL2::Core INTERFACE IMPORTED)
+    set_target_properties(SDL2::Core PROPERTIES
+        INTERFACE_COMPILE_OPTIONS
+        "SHELL:-s USE_SDL=2"
+        INTERFACE_LINK_OPTIONS
+        "SHELL:-s USE_SDL=2"
+    )
```
Comment (McMartin) [inline cmake/rigel.cmake:13]:
```suggestion
    target_compile_options(SDL2::Core INTERFACE "SHELL:-s USE_SDL=2")
    target_link_options(SDL2::Core INTERFACE "SHELL:-s USE_SDL=2")
```
and similarly for `SDL2::Mixer` and `Boost::boost`.

File: cmake/rigel.cmake [hunk 4]
```diff
@@ -26,3 +26,22 @@ function(rigel_define_wasm_targets_for_dependencies)
         "SHELL:-s USE_BOOST_HEADERS=1"
     )
 endfunction()
+
+
+function(rigel_disable_warnings target)
+    if (MSVC)
+        target_compile_options(${target} PRIVATE
+            /w
```
Comment (McMartin) [inline cmake/rigel.cmake:88]: MSVC supports both `/` and `-` for flags, so you could use `-w` with all compilers.

File: src/CMakeLists.txt (hunk 1 above)
Comment (lethal-guitar) [reply src/CMakeLists.txt]: Good catch, thanks!

File: CMakeLists.txt [hunk 5]
```diff
@@ -19,6 +19,7 @@ option(WARNINGS_AS_ERRORS "Treat compiler warnings as errors" ON)
 ###############################################################################
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/Modules/")
+include("${CMAKE_SOURCE_DIR}/cmake/rigel.cmake/")
```
[the remark this answers is not in the document]
Comment (lethal-guitar) [reply CMakeLists.txt]: I'm surprised it was working like this 🤔

File: CMakeLists.txt (hunk 2 above)
Comment (lethal-guitar) [reply CMakeLists.txt:39]: Nice! 👍

File: cmake/rigel.cmake (hunk 3 above)
Comment (lethal-guitar) [reply cmake/rigel.cmake:13]: Yeah that looks much more readable, thanks!

File: cmake/rigel.cmake (hunk 4 above)
Comment (lethal-guitar) [reply cmake/rigel.cmake:88]: Sounds good

**Comment by lethal-guitar** [comment]
@McMartin thanks for the review, really nice suggestions!

**Pull request merged**